From: Geert Uytterhoeven <geert@linux-m68k.org>
Date: Wed, 25 Jun 2014 10:31:09 +0200
Subject: [PATCH 35/56] enic: Make dummy rfs functions inline to fix
 !CONFIG_RFS_ACCEL build
Patch-mainline: v3.17-rc1
Git-commit: 644a918d20336a7deaa81f675c3c2f25bf3dafbb
References: bsc#911105 FATE#317501

If CONFIG_RFS_ACCEL=n:

drivers/net/ethernet/cisco/enic/enic_main.c: In function 'enic_open':
drivers/net/ethernet/cisco/enic/enic_main.c:1603:2: error: implicit declaration of function 'enic_rfs_flw_tbl_init' [-Werror=implicit-function-declaration]
drivers/net/ethernet/cisco/enic/enic_main.c: In function 'enic_stop':
drivers/net/ethernet/cisco/enic/enic_main.c:1630:2: error: implicit declaration of function 'enic_rfs_flw_tbl_free' [-Werror=implicit-function-declaration]

Introduced in commit a145df23ef32c7b933875f334ba28791ee75766e ("enic: Add
Accelerated RFS support").

Dummy functions are provided, but their prototypes are missing, causing the
build failure.  Provide dummy static inline functions instead to fix this.

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Govindarajulu Varadarajulu <gvaradar@cisco.com>
Acked-by: Gary Ching-Pang Lin <glin@suse.com>
---
 drivers/net/ethernet/cisco/enic/enic_clsf.c |   10 ----------
 drivers/net/ethernet/cisco/enic/enic_clsf.h |    3 +++
 2 files changed, 3 insertions(+), 10 deletions(-)

--- a/drivers/net/ethernet/cisco/enic/enic_clsf.c
+++ b/drivers/net/ethernet/cisco/enic/enic_clsf.c
@@ -269,14 +269,4 @@ ret_unlock:
 	return res;
 }
 
-#else
-
-void enic_rfs_flw_tbl_init(struct enic *enic)
-{
-}
-
-void enic_rfs_flw_tbl_free(struct enic *enic)
-{
-}
-
 #endif /* CONFIG_RFS_ACCEL */
--- a/drivers/net/ethernet/cisco/enic/enic_clsf.h
+++ b/drivers/net/ethernet/cisco/enic/enic_clsf.h
@@ -14,6 +14,9 @@ void enic_rfs_flw_tbl_init(struct enic *
 void enic_rfs_flw_tbl_free(struct enic *enic);
 int enic_rx_flow_steer(struct net_device *dev, const struct sk_buff *skb,
 		       u16 rxq_index, u32 flow_id);
+#else
+static inline void enic_rfs_flw_tbl_init(struct enic *enic) {}
+static inline void enic_rfs_flw_tbl_free(struct enic *enic) {}
 #endif /* CONFIG_RFS_ACCEL */
 
 #endif /* _ENIC_CLSF_H_ */
